src/Stack/Types/Config/Build.hs

```diff
@@ -167,6 +171,7 @@ instance FromJSON (WithJSONWarnings BuildOptsMonoid) where
               buildMonoidHaddockOpts <- jsonSubWarnings (o ..:? buildMonoidHaddockOptsArgName ..!= mempty)
               buildMonoidOpenHaddocks <- First <$> o ..:? buildMonoidOpenHaddocksArgName
               buildMonoidHaddockDeps <- First <$> o ..:? buildMonoidHaddockDepsArgName
+              buildMonoidHaddockInternal <- First <$> o ..:? buildMonoidHaddockInternalArgName
```

Comment:
Here is the line where the "haddock-internal" option is parsed from a `stack.yaml` file.